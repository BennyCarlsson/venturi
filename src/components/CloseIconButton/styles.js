import styled from "styled-components";
import { ReactComponent as XIcon } from "assets/icons/chevron-down.svg";
import Colors from "tokens/colors";

export const CloseIconButtonWrapper = styled.button`
  height: 38px;
  width: 38px;
  border-radius: 4px;
  padding: 0;
  border: none;
  background-color: ${Colors.vBlue};
  cursor: pointer;
`;

export const StyledXIcon = styled(XIcon)`
  stroke: ${Colors.white};
  transform: rotate(90deg);
`;
