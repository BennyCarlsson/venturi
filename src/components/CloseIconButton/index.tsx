import { CloseIconButtonWrapper, StyledXIcon } from './styles'

type CloseIconButtonProps = {
  onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
}

const CloseIconButton = ({ onClick, ...props }: CloseIconButtonProps) => {
  return (
    <CloseIconButtonWrapper onClick={onClick} {...props}>
      <StyledXIcon />
    </CloseIconButtonWrapper>
  )
}

export default CloseIconButton
