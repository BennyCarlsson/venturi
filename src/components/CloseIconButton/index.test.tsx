import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import CloseIconButton from '.';

describe('<CloseIconButton />', () => {
  const onClick = jest.fn();
  it('should match snapshot', () => {
    const tree = renderer.create(<CloseIconButton onClick={onClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    render(<CloseIconButton onClick={onClick} />);
    expect(screen).toBeTruthy();
  });

  it('should call onClick', () => {
    render(<CloseIconButton onClick={onClick} />);
    expect(onClick).not.toHaveBeenCalled();
    userEvent.click(screen.getByRole('button'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
