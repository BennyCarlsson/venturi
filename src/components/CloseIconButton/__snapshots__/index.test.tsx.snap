// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseIconButton /> should match snapshot 1`] = `
<button
  className="sc-bdfBwQ byoPcS"
  onClick={[MockFunction]}
>
  <svg
    className="sc-gsTCUz iynNxB"
  >
    chevron-down.svg
  </svg>
</button>
`;
